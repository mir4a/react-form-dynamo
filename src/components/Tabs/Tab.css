.Tab {
  display: flex;
  flex: 1;
  flex-direction: column;
  text-align: left;
  border: 2px solid var(--btn-primary);
  padding: 15px;
}