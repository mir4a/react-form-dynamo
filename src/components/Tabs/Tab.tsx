import React, { ReactNode } from 'react';

interface Props {
  isActive?: boolean;
  children: ReactNode;
}

export default function Tab(props: Props) {
  return (
    <>
      is Active?: {JSON.stringify(props.isActive)}
      <>{props.children}</>
    </>
  );
}
