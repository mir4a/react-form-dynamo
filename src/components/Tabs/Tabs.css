.Tabs {
  display: flex;
  flex-direction: column;
  margin: 20px;
  min-height: 300px;
  width: 100%;
  max-width: 900px;
}

@media (max-width: 1000px) {
  .Tabs {
    width: auto;
    max-width: 100%;
  }
}

.Tabs-wrapper {
  display: flex;
  flex: 1;
  width: 100%;
}

.Tabs-navigation {
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap-reverse;
  position: relative;
  bottom: -3px;
}

.Tabs-button {
  font-size: 16px;
  font-weight: 700;
  color: white;
  background-color: var(--btn-primary);
  border: 2px solid var(--btn-primary);
  border-bottom-width: 0;
  padding: 10px 20px
}

.Tabs-button:active,
.Tabs-button:focus {
  outline: none;
}

.Tabs-button:hover {
  cursor: pointer;
  border-color: var(--btn-hover);
  background-color: var(--btn-hover);
}

.Tabs-button__active {
  color: var(--text-color);
  background-color: white;
}

.Tabs-button__active {
  cursor: default;
  pointer-events: none;
}