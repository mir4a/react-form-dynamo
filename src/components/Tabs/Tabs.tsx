import React, { useState, useCallback, ReactNode, useMemo } from 'react';
import TabContent from './Tab';

type Tab = {
  isActive?: boolean;
  name: string;
  content: ReactNode;
};

interface Props {
  tabs: Tab[];
}

export function TabButton(props: {
  name: string;
  isActive?: boolean;
  onClick: (e: React.MouseEvent) => void;
}) {
  const { name, isActive, onClick } = props;
  return (
    <>
      <button onClick={onClick}>
        isActive?: {JSON.stringify(isActive)},{name}
      </button>
    </>
  );
}

export default function Tabs(props: Props) {
  const { tabs } = props;
  const [activeTab, setActiveTab] = useState<number | undefined>();

  const [tabButtons, tabItems] = useMemo(() => {
    let tabButtons: any[] = [];
    let tabItems: any[] = [];
    tabs.forEach((tab, index) => {
      tabButtons.push(
        <TabButton
          key={`TabButton_${index}`}
          name={tab.name}
          isActive={activeTab ? index === activeTab : tab.isActive}
          onClick={() => {
            setActiveTab(index);
          }}
        />,
      );

      tabItems.push(
        <TabContent key={`TabContent_${index}`} isActive={tab.isActive}>
          {tab.content}
        </TabContent>,
      );
    });

    return [tabButtons, tabItems];
  }, [tabs, activeTab]);

  const handleTabChange = useCallback(
    function changeTab(tabId: number): void {
      setActiveTab(tabId);
    },
    [setActiveTab],
  );

  return (
    <>
      <h1>This is tabs:</h1>
      <div>{tabButtons}</div>
      <hr />
      <div>{activeTab ? tabItems![activeTab] : tabItems[0]}</div>
    </>
  );
}
