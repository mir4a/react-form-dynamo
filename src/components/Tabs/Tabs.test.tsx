import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './Tabs';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Tabs
        tabs={[
          { isActive: true, name: 'Tab 1', content: <div>Content 1</div> },
          { name: 'Tab 2', content: <span>Content 2</span> },
        ]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with empty tabs', () => {
  const tree = renderer.create(<Tabs tabs={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
