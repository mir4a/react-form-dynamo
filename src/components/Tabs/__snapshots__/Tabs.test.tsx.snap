// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Tabs"
>
  <div
    className="Tabs-navigation"
  >
    <button
      className="Tabs-button Tabs-button__active"
      onClick={[Function]}
    >
      Tab 1
    </button>
    <button
      className="Tabs-button"
      onClick={[Function]}
    >
      Tab 2
    </button>
  </div>
  <div
    className="Tabs-wrapper"
  />
</div>
`;

exports[`renders correctly with empty tabs 1`] = `
<div
  className="Tabs"
>
  <div
    className="Tabs-navigation"
  />
  <div
    className="Tabs-wrapper"
  />
</div>
`;
